### PR TITLE
Completed sgt-back-end challenge. - Daniel Tak

### PR DESCRIPTION
**Pull request by d-tak**
**GET**
Screen Shot 2022-08-25 at 4 50 15 PM

**POST**
Screen Shot 2022-08-25 at 9 33 11 PM
Screen Shot 2022-08-25 at 9 22 27 PM
Screen Shot 2022-08-26 at 10 48 02 AM
Screen Shot 2022-08-26 at 10 48 55 AM

**PUT**
Screen Shot 2022-08-26 at 10 50 49 AM
Screen Shot 2022-08-26 at 10 50 26 AM
Screen Shot 2022-08-26 at 10 50 02 AM

**DELETE**
Screen Shot 2022-08-26 at 10 51 22 AM
Screen Shot 2022-08-26 at 10 22 55 AM
Screen Shot 2022-08-26 at 10 32 50 AM

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    });
+});
+
+// POST (inserts a new grade and returns created grade)
+app.use(express.json());
+app.post('/api/grades', (req, res) => {
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    res.status(400);
+    res.json({ error: 'Name, Course, AND Score are REQUIRED fields' });
+    return;
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:39]: A `score` of `0` should not result in a `400 Bad Request`

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by d-tak** [comment]
Screen Shot 2022-08-26 at 11 45 57 AM

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+app.put('/api/grades/:gradeId', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const gradeId = Number(req.params.gradeId);
+
+  if (!name || !course || !score) {
+    res.status(400);
+    res.json({ error: 'Name, Course, AND Score are REQUIRED fields' });
+    return;
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:99]: A `score` of `0` should not trigger a `400 Bad Request`.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by d-tak** [comment]
Screen Shot 2022-08-26 at 11 54 13 AM

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**